lib/pf/api.pm

```diff
@@ -222,9 +223,29 @@ sub firewallsso : Public {
 
     my $logger = pf::log::get_logger();
 
-    foreach my $firewall_conf ( sort keys %pf::config::ConfigFirewallSSO ) {
-        my $firewall = pf::factory::firewallsso->new($firewall_conf);
-        $firewall->action($firewall_conf,$postdata{'method'},$postdata{'mac'},$postdata{'ip'},$postdata{'timeout'});
+    foreach my $firewall_id ( sort keys %pf::config::ConfigFirewallSSO ) {
+        my $firewall = pf::factory::firewallsso->new($firewall_id);
+
+        if($postdata{method} eq "Update"){
+            if( pf::util::isenabled($pf::config::ConfigFirewallSSO{$firewall_id}{cache_updates}) ){
+                my $cache = pf::CHI->new( namespace => 'firewall_sso' );
+                my $cache_timeout = $pf::config::ConfigFirewallSSO{$firewall_id}{cache_timeout} || $postdata{timeout} / 2;
```

Comment:
By multiplying by 2 i was thinking on the registration/isolation vlan were we have a dhcp request every 30s.
So maybe we have to imagine to have a threshold on $postdata{timeout} like if < 300s then use the default $DEFAULT_LEASE_LENGTH. So the first request on the reg/isol vlan will trigger the firewall sso and not the hundred other request after.